test/core/host_api/child_storage_extension_test.cpp

```diff
@@ -0,0 +1,494 @@
+
+  // 'func' (lambda)
+
+  EXPECT_CALL(*trie_child_storage_batch_, get(key))
+      .WillOnce(Return(GetParam()));
+
+  // results
+
+  ASSERT_EQ(value_span,
+            child_storage_extension_->ext_default_child_storage_get_version_1(
+                child_storage_key_span, key_span));
+}
+
+// ext_default_child_storage_read_version_1
+TEST_P(ReadOutcomeParameterizedTest, ReadTest) {
+  // executeOnChildStorage
+
+  WasmPointer child_storage_key_pointer = 42;
+  WasmSize child_storage_key_size = 42;
+  WasmSpan child_storage_key_span =
+      PtrSize(child_storage_key_pointer, child_storage_key_size).combine();
+  Buffer child_storage_key(8, 'l');
+  WasmPointer key_pointer = 43;
+  WasmSize key_size = 43;
+  WasmSpan key_span = PtrSize(key_pointer, key_size).combine();
+  Buffer key(8, 'k');
+  EXPECT_CALL(*memory_,
+              loadN(child_storage_key_pointer, child_storage_key_size))
+      .WillOnce(Return(child_storage_key));
+  EXPECT_CALL(*memory_, loadN(key_pointer, key_size)).WillOnce(Return(key));
+
+  if (GetParam()) {
+    RootHash new_child_root = "123456"_hash256;
+    Buffer new_child_root_buffer{scale::encode(new_child_root).value()};
+    EXPECT_CALL(*trie_child_storage_batch_, commit())
+        .WillOnce(Return(new_child_root));
+
+    Buffer prefixed_child_storage_key =
+        Buffer{kagome::storage::kChildStorageDefaultPrefix}.putBuffer(
+            child_storage_key);
+    EXPECT_CALL(*trie_batch_,
+                put(prefixed_child_storage_key, new_child_root_buffer))
+        .WillOnce(Return(outcome::success()));
+  }
+
+  // logic
+
+  WasmPointer value_pointer = 44;
+  WasmSize value_size = 44;
+  WasmSpan value_span = PtrSize(value_pointer, value_size).combine();
+  auto encoded_result =
+      scale::encode<std::optional<uint32_t>>(std::nullopt).value();
+
+  WasmOffset offset = 4;
+  Buffer offset_value_data;
+
+  if (GetParam()) {
+    offset_value_data = GetParam().value().subbuffer(offset);
+    ASSERT_EQ(offset_value_data.size(), GetParam().value().size() - offset);
+    EXPECT_OUTCOME_TRUE(
+        encoded_opt_offset_val_size,
+        scale::encode(std::make_optional<uint32_t>(offset_value_data.size())));
+    encoded_result = encoded_opt_offset_val_size;
+    EXPECT_CALL(
+        *memory_,
+        storeBuffer(value_pointer, gsl::span<const uint8_t>(offset_value_data)))
+        .WillOnce(Return());
+  }
+
+  WasmSpan res_wasm_span = 1337;
+  EXPECT_CALL(*memory_, storeBuffer(gsl::span<const uint8_t>(encoded_result)))
+      .WillOnce(Return(res_wasm_span));
+
+  // 'func' (lambda)
+
+  EXPECT_CALL(*trie_child_storage_batch_, get(key))
+      .WillOnce(Return(GetParam()));
+
+  // results
+
+  ASSERT_EQ(res_wasm_span,
+            child_storage_extension_->ext_default_child_storage_read_version_1(
+                child_storage_key_span, key_span, value_span, offset));
+}
+
+// ext_default_child_storage_set_version_1
+TEST_P(VoidOutcomeParameterizedTest, SetTest) {
```

Comment:
Please, leave given-when-then doc comment for all tests

Reply:
Will add them.